clients/unshaded/src/main/java/tachyon/hadoop/AbstractTFS.java

```diff
@@ -309,20 +311,25 @@ public long getDefaultBlockSize() {
     for (int k = 0; k < blocks.size(); k ++) {
       FileBlockInfo info = blocks.get(k);
       long offset = info.getOffset();
-      long end = offset + info.getLength();
+      long end = offset + info.blockInfo.getLength();
       // Check if there is any overlapping between [start, start+len] and [offset, end]
       if (end >= start && offset <= start + len) {
         ArrayList<String> names = new ArrayList<String>();
         ArrayList<String> hosts = new ArrayList<String>();
-        for (NetAddress addr : info.getLocations()) {
+        List<NetAddress> addrs = Lists.newArrayList(info.getUnderFsLocations());
+        // add the existing in-memory block locations
+        for (tachyon.thrift.BlockLocation location : info.getBlockInfo().getLocations()) {
```

Comment:
Does this mean the ufs locations are added first, then tachyon locations? Tachyon locations should come first.